parlai/core/teachers.py

```diff
@@ -1944,6 +1945,223 @@ def shutdown(self):
             t.shutdown()
 
 
+class ChunkTeacher(FixedDialogTeacher, ABC):
```

Comment:
We could follow Mary and call it AbstractChunkTeacher. I'm ambivalent.

Reply:
I'm on the fence about that too - after all, TorchRankerAgent has an abstract method and it's not AbstractTorchRankerAgent. I think I'm leaning against the "Abstract" for conciseness, since it's clear from the inheritance of ABC that it's abstract